unified-codes/libs/ui/src/lib/components/entity/molecules/EntityTable.tsx

```diff
@@ -1,88 +1,56 @@
+import { Grid } from '../../layout/atoms';
 
-import EntityTableHeader, { EntityTableHeaderProps } from './EntityTableHeader';
-import EntityTableRow, { EntityTableRowProps } from './EntityTableRow';
-export interface EntityTableProps {
-  bodyProps?: TableBodyProps;
-  data: Array<IEntity>;
-  headProps?: TableHeadProps;
-  headerProps?: EntityTableHeaderProps;
-  rowProps?: EntityTableRowProps;
-  tableProps?: TableProps;
-  variables?: IExplorerVariables;
-  onSort?: (value: string) => void;
-  onEntitySelect: (entityCode: string) => void;
+export interface IEntityTableProps {
+  classes?: {
+    body?: string,
+    head?: string,
+    paginationContainer?: string,
+    root?: string,
+    tableContainer?: string
+  };
+  header: React.ReactElement,
+  pagination: React.ReactElement,
+  rows: React.ReactElement,
 }
 
-export type EntityTable = React.FunctionComponent<EntityTableProps>;
+export type EntityTable = React.FunctionComponent<IEntityTableProps>;
 
 export const EntityTable: EntityTable = ({
-  tableProps,
-  headProps,
-  headerProps,
-  bodyProps,
-  rowProps,
-  data,
-  variables,
-  onSort,
-  onEntitySelect,
-}: EntityTableProps) => {
-  const mapEntity = (entity: IEntity, index: number) => {
-    const localRowProps =
-      tableProps?.stripedRows && tableProps?.alternatingRowColour
-        ? { style: { backgroundColor: index % 2 ? tableProps?.alternatingRowColour : '', cursor: 'pointer' } }
-        : undefined;
-
-    return (
-      <EntityTableRow
-        key={entity.code}
-        {
-          ...{
-            ...rowProps,
-            entity,
-            onEntitySelect, 
-            rowProps: {
-              ...localRowProps,
-              hover: true
-            },
-          }
-        }
-        
-      ></EntityTableRow>
-    );
-  };
-
-  const EntityTableRows = React.useCallback(
-    () => <React.Fragment>{data.map(mapEntity)}</React.Fragment>,
-    [rowProps, data]
-  );
-  const { stripedRows, alternatingRowColour, ...otherProps } = tableProps || {};
+  classes,
+  header,
+  pagination,
+  rows,
+}: IEntityTableProps) => {
+ 
   return (
-    <Table {...otherProps}>
-      <TableHead {...headProps}>
-        <EntityTableHeader
-          {...headerProps}
-          onSort={onSort}
-          orderDesc={variables?.orderDesc}
-          orderBy={variables?.orderBy}
-        />
-      </TableHead>
-      <TableBody {...bodyProps}>
-        <EntityTableRows />
-      </TableBody>
-    </Table>
+    <Grid container classes={{ root: classes?.root }}>
+      <Grid container item classes={{ root: classes?.tableContainer }}>
+        <Table classes={{ root: classes?.root }}>
+          <TableHead classes={{ root: classes?.head }}>
+            {header}
+          </TableHead>
+          <TableBody classes={{ root: classes?.body }}>
+            {rows}
+          </TableBody>
+          <TableFooter classes={{ root: classes?.paginationContainer }}>
+            <TableRow>
```

Comment:
Ah, OK!

It's odd because the intended usage from the material docs (https://material-ui.com/api/table-pagination/#tablepagination-api) suggests that pagination has to be inside the table context (inside a row), and I get warning errors in the console when I had it outside of a `TableRow` context. On the other hand, the material UI docs have an example which contradicts this: https://codesandbox.io/s/yi98d?file=/demo.tsx.

I'm thinking it would be nice to fix the header and footer and only scroll the "content area", in this case the table. But I agree 100% that pagination should be sticky, just like the header. Will experiment and see if I can find a solution (if turns out to be complicated, will create an issue!).

Reply:
Also have discovered that there's an issue with scrolling with the toggle bar. The design has the content area "overlapping" the header but also blending into it so it appears as a single component. Finding it quite tricky to design this in a nice compositional way!

Reply:
yeah, on both counts. My thinking is to go with whatever is cleanest in the code and vaguely sensible for the UI.
I spent a fair bit of time beating the page height etc into shape in order to get the footer and pagination sticky, and I felt it was a bit hacky while doing so.

Would be better to talk about what is nicely usable and just run with that. which is all to say - it's ok like it is, and don't spend too much time trying to get it perfect!

Reply:
This has done the job for me!

58c0690c7331c3d99eb57baf983c99f5d297b472.